data/src/main/java/com/linkedin/data/schema/DataSchemaParser.java

```diff
@@ -0,0 +1,114 @@
+package com.linkedin.data.schema;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.util.List;
+
+/**
+ * A schema parser is used to parse data schemas for a particular source representation.
+ */
+public interface DataSchemaParser {
```

Comment:
Would you mind sharing the use case where you need to provide another implementation of DataSchemaParser that is different from current SchemaParser implementation?